ui/models/job.js

```diff
@@ -76,7 +76,13 @@ export default class JobModel {
   static get(repoName, pk, signal) {
     // a static method to retrieve a single instance of JobModel
     return fetch(`${uri}${pk}/`, { signal })
-      .then(response => response.json().then(data => new JobModel(data)));
+      .then(async (response) => {
+        if (response.ok) {
+          const job = await response.json();
+          return new JobModel(job);
+        }
+        throw Error(`Error loading job with id: ${pk} : ${response.statusText}`);
```

Comment:
Maybe replace:
`with id: ${pk} :`
...with:
`with id ${pk}:` ?

Reply:
I wonder if it'd be more useful to return the response message for errors rather than statusText, which would basically be a somewhat cryptic 'bad request', (etc)?

Reply:
Good call on both counts.  Fixed.  :)